assets/scripts/segments/drag_and_drop.js

```diff
@@ -215,6 +219,46 @@ export function onBodyMouseDown (event) {
   }
 }
 
+export function isSegmentWithinCanvas (event, canvasEl) {
+  const { remainingWidth } = store.getState().street
+
+  let x, y
+  if (event.touches && event.touches[0]) {
+    x = event.touches[0].pageX
+    y = event.touches[0].pageY
+  } else {
+    x = event.x
+    y = event.y
+  }
+
+  // For some reason, the last drag event of a palette segment causes x and
+  // y to be reset to 0 despite the last mouse position not being (0,0). The
+  // code below returns true if the second to last drag event was within canvas.
+  // TODO - find reason why last drag event resets x, y
+  if (x === 0 && y === 0 && oldDraggingState.withinCanvas) return true
```

Comment:
This is really interesting. Some questions to explore:

- Is this true across browsers? (If not, it might be a browser-specific quirk.)
- What happens if you use [`event.clientX`](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/clientX) over `event.x`? Are they the same?
- Perhaps the palette segment element is being destroyed/removed just prior to the last drag event? Is there an order of operations thing causing this?

Regardless of whether we ever find out, if the solution is working consistently, then 👍

Reply:
`event.clientX` gives me the same value as `event.x`. I don't think it has to do with the order of operations because I just realized it only occurs when I'm hovering the palette segment over the street name canvas... Do you think it has something to do with the `<StreetNameCanvas /`> being of a higher z-index than `<StreetEditable />` and me attaching the listener inside `<StreetEditable/>`?

Reply:
Yea, it looks like it's because of the `<StreetNameCanvas />` because if I set the z-index to 0, making it invisible, then the bug no longer occurs.

Reply:
Cool! One thing we could try is to find the top-level DOM node for `<StreetNameCanvas />`, get its class name, then write (in CSS):

```css
body.segment-move-dragging .class-name-for-street-name-canvas-root-node {
    pointer-events: none;
}
```

This forces the element not to capture mouse events while we're in dragging state, but it won't alter its z-index or visibility.

Reply:
Ah yes! That works. 
Actually now that you mention it, for the classNames that we add dynamically to the body when we drag (such as `segment-move-dragging`, `within-canvas`, and `not-within-canvas`) is there anything preventing us from now adding those classes to the segment itself?